app/src/main/java/ch/sdp/vibester/activity/DeleteSongsActivity.kt

```diff
@@ -0,0 +1,145 @@
+package ch.sdp.vibester.activity
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.os.Environment
+import android.util.Log
+import android.view.Gravity
+import android.view.View
+import android.view.ViewGroup.LayoutParams
+import android.view.Window
+import android.widget.Button
+import android.widget.LinearLayout
+import android.widget.LinearLayout.LayoutParams.MATCH_PARENT
+import android.view.ViewGroup.LayoutParams.MATCH_PARENT
+import android.view.ViewGroup.LayoutParams.WRAP_CONTENT
+import android.widget.TextView
+import android.widget.Toast
+import ch.sdp.vibester.R
+import org.w3c.dom.Text
+import java.io.*
+
+class DeleteSongsActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        requestWindowFeature(Window.FEATURE_NO_TITLE)
+        supportActionBar?.hide()
+
+        setContentView(R.layout.activity_delete_songs)
+
+        val layout: LinearLayout = findViewById(R.id.delete_songs_linear)
+        generateButtons(layout)
+    }
+
+    /*
+     * Generates buttons for each downloaded extract to be added to the scrollview layout.
+     * Assigns each button a call to the "deleteDownloadedSong" function as a listener.
+     */
+    private fun generateButtons(layout: LinearLayout) {
+        var records = File(applicationContext.getExternalFilesDir(Environment.DIRECTORY_DOWNLOADS), "records.txt")
+
+        if(!records.exists() || records.length() == 0L) {
+            createNoSongsView(layout)
+        } else {
+            var reader = BufferedReader(FileReader(records))
+            var currentLine = reader.readLine()
+            var iterator = 0
+
+            while(currentLine != null) {
+                var trimmed = currentLine.trim()
+                if(trimmed.isNotEmpty()) {
+                    val deleteButton = createButton(trimmed, iterator, layout)
+                    layout.addView(deleteButton)
+                }
+                currentLine = reader.readLine()
+                iterator += 1
+            }
+            reader.close()
+        }
+    }
+
+    /*
+     * Handles the creation and initialization of buttons.
+     */
+    private fun createButton(trimmed: String, iterator: Int, layout: LinearLayout): Button {
+        val deleteButton = Button(this)
+        deleteButton.id = iterator //resources.getIdentifier("id_$trimmed", "id", packageName)
```

Comment:
what is the purpose of the comment? useful to keep?

Reply:
Previously was the commented one which I wanted to keep. However the getId method always returned 0 no matter what, so I kept it as a comment. Good catch, will remove it!